sbin/pf-mariadb

```diff
@@ -173,16 +175,22 @@ Test the connection to the database for a specific host using the replication cr
 
 sub test_db {
     my ($host) = @_;
-    my $mydbh = DBI->connect( "dbi:mysql:dbname=mysql;host=$host;port=3306",
-        $Config{active_active}{galera_replication_username}, $Config{active_active}{galera_replication_password}, { RaiseError => 0, PrintError => 0, mysql_auto_reconnect => 1 } );
-
-    # make sure we have a database handle
-    if ($mydbh) {
-        return 1;
-    }
-    else {
-        return 0;
-    }
+    eval {
+        local $SIG{ALRM} = sub {die "Timeout connecting to MySQL for host $host\n";};
+        alarm($MYSQL_CONNECT_TIMEOUT);
+        my $mydbh = DBI->connect( "dbi:mysql:dbname=mysql;host=$host;port=3306",
```

Comment:
There is a built in option to do this mysql_connect_timeout.

http://search.cpan.org/~capttofu/DBD-mysql-4.023/lib/DBD/mysql.pm#Class_Methods

This should be more reliable since it is handled in mysql client instead using unix signals.

Reply:
fixaide